data_steward/cdr_cleaner/manual_cleaning_rules/dc732_suppress.py

```diff
@@ -0,0 +1,445 @@
+    :return: dataframe with dataset_id, table, disallowed_concept_id, person_id, row_id, source
+    """
+    summary_cols = ['dataset_id', 'table', 'disallowed_concept_id', 'person_id', 'row_id', 'source']
+    group_by_concept = rows_to_suppress_df[summary_cols].groupby(['dataset_id', 'table', 'disallowed_concept_id'])
+    summary_df = group_by_concept.nunique().drop(columns=['dataset_id', 'table', 'disallowed_concept_id'])
+    summary_df = summary_df.add_prefix('n_')
+    summary_df['ppi_pm'] = group_by_concept.apply(ppi_pm)
+    return summary_df.sort_values(by=['dataset_id', 'table', 'n_person_id'], ascending=False)
+
+
+# # Backup Rows
+# Backup rows are stored in tables whose structure is identical to domain tables except for the addition of fields to distinguish the dataset and table the original row came from. For example, `visit_occurrence` rows would be backed up in a table with the below schema.
+#
+# | Field name                    | Type    |
+# |-------------------------------|---------|
+# | **dataset_id**                | STRING  |
+# | **table**                     | STRING  |
+# | visit_occurrence_id           | INTEGER |
+# | person_id                     | INTEGER |
+# | visit_concept_id              | INTEGER |
+# | --- _other columns_  ---      |         |
+# | discharge_to_source_value     | STRING  |
+# | preceding_visit_occurrence_id | INTEGER |
+
+# +
+BACKUP_TABLE_ROWS_QUERY = JINJA_ENV.from_string("""
+{% for dataset_id in dataset_ids %}
+SELECT 
+ '{{dataset_id}}' AS dataset_id, 
+ '{{table}}' table, 
+ t.* 
+FROM {{dataset_id}}.{{table}} t
+JOIN {{suppress_rows_table}} r
+ ON r.row_id = t.{{table}}_id
+WHERE r.dataset_id = '{{dataset_id}}'
+AND r.table = '{{table}}'
+{% if loop.nextitem is defined %}
+ UNION ALL
+{% endif %}
+{% endfor %}
+""")
+
+BACKUP_EXT_TABLE_ROWS_QUERY = JINJA_ENV.from_string("""
+{% for dataset_id in dataset_ids %}
+SELECT 
+ '{{dataset_id}}' AS dataset_id, 
+ '{{table}}_ext' table, 
+ t.* 
+FROM {{dataset_id}}.{{table}}_ext t
+JOIN {{suppress_rows_table}} r
+ ON r.row_id = t.{{table}}_id
+WHERE r.dataset_id = '{{dataset_id}}'
+AND r.table = '{{table}}'
+{% if loop.nextitem is defined %}
+ UNION ALL
+{% endif %}
+{% endfor %}
+""")
+
+
+def backup_rows_to_suppress(tables_to_suppress_df, suppress_rows_table):
+    """
+    Create a table for each domain and load rows that will be suppressed across all datasets
+    Note: Rows are appended if tables exist
+    
+    :param tables_to_suppress_df: dataframe with dataset_id, table_name, ext_table_name, columns
+    :param suppress_rows_table: identifies the table where suppress rows are loaded
+    :return: list of query jobs
+    """
+    tables_group_df = tables_to_suppress_df.groupby(by=['table_name', 'ext_table_name'])
+    query_jobs = []
+    for (table_name, ext_table_name), table_info in tables_group_df:
+        dataset_ids = table_info.dataset_id.unique()
+        if table_name != 'observation':
```

Comment:
I don't understand.  Why don't we back up rows from the observation tables?

Reply:
Woah are you right- this is an ad-hoc thing I didn't intend to push. Will fix. Thanks!